include/rocksdb/perf_context.h

```diff
@@ -14,34 +14,288 @@
 
 namespace ROCKSDB_NAMESPACE {
 
+/*
+ * Please add new metrics to this macro and appropriate fields will be copied,
+ * and/or emitted when converted to string.
+ * When people need to add new metrics please add an optional multi-line C style
+ * comment on what the metric semantics are and enclose the specific metric
+ * within defCmd()
+ */
+#define DEF_PERF_CONTEXT_LEVEL_METRICS(defCmd)                                   \
+  /* # of times bloom filter has avoided file reads, i.e., negatives. */         \
+  defCmd(bloom_filter_useful) /* # of times bloom FullFilter has not avoided     \
+                                 the reads. */                                   \
+      defCmd(bloom_filter_full_positive)      /*                                 \
+                                               * # of times bloom FullFilter has \
+                                               * not avoided the reads and data  \
+                                               * actually      exist.            \
+                                               */                                \
+      defCmd(bloom_filter_full_true_positive) /*                                 \
+                                               * total number of user key        \
+                                               * returned (only include keys     \
+                                               * that are found, does not        \
+                                               * include keys that are deleted   \
+                                               * or merged without a final put   \
+                                               */                                \
+      defCmd(user_key_return_count) /* total nanos spent on reading data from    \
+                                       SST files */                              \
+      defCmd(get_from_table_nanos)                                               \
+          defCmd(block_cache_hit_count) /* total number of block cache hits */   \
+      defCmd(block_cache_miss_count)    /* total number of block cache misses */
```

Comment:
While I love the line count of this PR (net removal of 462 lines), the refactoring part of this PR opposes our style guide: https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros. The part of that I care about is making sure readers of the public API - not just the compiler - can easily decipher what is in the public API. I don't care about avoiding macros entirely; we use them in various places like "monitoring/perf_context.cc" as you probably noticed.

Is there a way to keep the public header the same and iterate over the `uint64_t`s in the various places we need to do something affecting every field? We can introduce a rule in `PerfContext` that it can only contain `uint64_t` members. I think this way can still eliminate the duplication except for `PerfContext::ToString()`'s implementation.

Reply:
thanks for taking a look, Andrew! yeah you are right. My bad that I missed a coding style guide. Let me think of a way to fix this and get back to you